src/pytorch_lightning/core/saving.py

```diff
@@ -171,10 +171,10 @@ def on_hpc_load(self, checkpoint: Dict[str, Any]) -> None:
 
 
 def _load_from_checkpoint(
-    cls: Union["pl.LightningModule", "pl.LightningDataModule"],
-    checkpoint_path: Union[str, IO],
+    cls: Union["pl.LightningModule", Type["pl.LightningDataModule"]],
```

Comment:
updated to resolve

```sh
src/pytorch_lightning/core/datamodule.py:248: error: Argument 1 to "_load_from_checkpoint" has incompatible type "Type[LightningDataModule]"; expected "Union[LightningModule, LightningDataModule]"  [arg-type]
src/pytorch_lightning/core/datamodule.py:249: error: Argument 2 to "_load_from_checkpoint" has incompatible type "Union[Union[str, Path], IO[Any]]"; expected "Union[str, IO[Any]]"  [arg-type]
src/pytorch_lightning/core/datamodule.py:251: error: Argument "hparams_file" to "_load_from_checkpoint" has incompatible type "Union[str, Path, None]"; expected "Optional[str]"  [arg-type]
```

Reply:
This is absolutely not correct - either both should be `Type[...]` or instances